jwst/datamodels/container.py

```diff
@@ -530,7 +464,15 @@ def get_crds_parameters(self):
         Returns
         -------
         dict
+
+        Notes
+        -----
+        stpipe requires ModelContainer to have a crds_observatory attribute in order
+        to pass through step.run(), but it is never accessed.
         """
+        msg = ("stpipe uses the get_crds_parameters method from the 0th model in the "
+               "ModelContainer. This method is currently not used.")
+        raise NotImplementedError(msg)
         with self._open_first_science_exposure() as model:
             return model.get_crds_parameters()
 
```

Comment:
The code below the NotImplementedError is unreachable.  If we want to keep it around in case it's useful in the future, maybe just warn and remove the NotImplementedError?  If we don't think the implementation is right at all, we should just remove it.

Reply:
If we do keep `_open_first_science_exposure`, it needs a small fix to remove the self.meta reference.

Reply:
Sorry, looking at it more - `_open_first_science_exposure` just looks fragile and unusable as is.  I say we remove it and just keep the NotImplementedError.